inference-engine/src/cldnn_engine/cldnn_engine.cpp

```diff
@@ -126,24 +139,72 @@ InferenceEngine::ICNNNetwork::Ptr clDNNEngine::CloneAndTransformNetwork(const In
+            manager.register_pass<ngraph::pass::ConvertPriorBox>();
+            manager.register_pass<ngraph::pass::CommonOptimizations>();
+            manager.register_pass<ngraph::pass::ConvertOpSet3ToOpSet2>();
+            manager.register_pass<ngraph::pass::ConvertOpSet2ToOpSet1>();
+
+
+            manager.set_callback(transformations_callback);
+            manager.run_passes(nGraphFunc);
+
+            const auto fp16_callback = [&baselineIsFP16](const std::shared_ptr<const ::ngraph::Node> &node) -> bool {
+                if (!baselineIsFP16 && node->get_output_element_type(0) == ngraph::element::f16) {
+                    baselineIsFP16 = true;
+                }
+
+                return true;
+            };
+
+            ngraph::pass::Manager conversion_manager;
+
+            if (enableInt8) {
+                // [WA part1] Convert quantized FP16 model to FP32 to avoid possible overflow and mixed precision errors
+                conversion_manager.register_pass<ngraph::pass::ConvertPrecision>(ngraph::element::f16, ngraph::element::f32);
+            }
+
+            conversion_manager.set_callback(fp16_callback);
+            conversion_manager.run_passes(nGraphFunc);
+
+            ngraph::pass::Manager ti_manager;
+            // Unroll will be called after all conversions
+            // temporarily switch back to plugin unroller from NGraph unroller until TI output names are corrected
+            // ti_manager.register_pass<ngraph::pass::UnrollTensorIterator>();
+            ti_manager.run_passes(nGraphFunc);
+        }
+
+        using namespace ngraph::pass::low_precision;
+        if (enableInt8) {
+            auto params = LayerTransformation::Params(
+                true,  // updatePrecisions
+                LayerTransformation::QuantizedTensorAlignment::UpdateLevel,  // quantizedTensorAlignmentOnActivations
+                LayerTransformation::QuantizedTensorAlignment::None,  // quantizedTensorAlignmentOnWeights
+                true);  // supportAsymmetricQuantization
+            LowPrecisionTransformer transformer(LowPrecisionTransformer::getAllTransformations(params)
+                .add<MatMulTransformation, ngraph::opset1::MatMul>(LayerTransformation::Params(params).setSupportAsymmetricQuantization(false)));
+
+            transformer.transform(nGraphFunc);
+        }
+
+        {
+            ngraph::pass::Manager manager = ngraph::pass::Manager();
+            manager.register_pass<ngraph::pass::ConvertOpSet1ToLegacy>();
+            manager.set_callback(transformations_callback);
+            manager.run_passes(nGraphFunc);
+
+            ngraph::pass::Manager ti_manager;
+            // Unroll will be called after all conversions
+            // temporarily switch back to plugin unroller from NGraph unroller until TI output names are corrected
+            // ti_manager.register_pass<ngraph::pass::UnrollTensorIterator>();
+            ti_manager.run_passes(nGraphFunc);
```

Comment:
Do we really need to run UnrollTensorIterator pass twice? (I know that it's disabled right now, but the code is duplicated)